firedrake/functionspaceimpl.py

```diff
@@ -822,3 +827,75 @@ def ComponentFunctionSpace(parent, component):
     new.component = component
     new.parent = parent
     return new
+
+
+class RealFunctionSpace(FunctionSpace):
+    """:class:`FunctionSpace` based on elements of family "Real". A
+    :class`RealFunctionSpace` only has a single global value for the
+    whole mesh.
+
+    This class should not be directly instantiated by users. Instead,
+    FunctionSpace objects will transform themselves into
+    :class:`RealFunctionSpace` objects as appropriate.
+
+    """
+
+    def __init__(self, mesh, element, name):
+
+        self.name = name
+        self._index = None
+        self._ufl_element = element
+        self.fiat_element = None
+        self._mesh = mesh
+        self.dim = 1
+        self.rank = 0
+
+        self._node_count = 1
+
+        self.comm = mesh.comm
+
+        """:class:`RealFunctionSpace` objects have no node set."""
+        self.node_set = None
```

Comment:
Most of these are argument independent, so I would make them class attributes:

```python
class RealFunctionSpace(FunctionSpace):

    finat_element = None
    dim = 1
    rank = 0
    shape = ()
    node_set = None

    def __init__(self, mesh, element, name):
        self._ufl_element = element
        self.name = name
        self.comm = mesh.comm
        self._mesh = mesh
        self.dof_dset = op2.GlobalDataSet(self.make_dat())
```

Note as well we now have `finat_element`, not `fiat_element`.